### PR TITLE
[DT 1133] Add fallback mechanism for RPC providers

### PR DESCRIPTION
**Pull request by tunguyennnnn**
## Background

Rpc endpoints rely entirely on externl providers. This add a fallback mechanism to different provider and fast failure on first provider to increase uptime.

## Changes
Add fallback mechanism:
* if first RPC provider call fail, it always fallback to second RPC provider calls
* if 70% of first RPC call fail, next requests within 20s will always invoke second RPC provider
* after 20s, it will retry with first RPC provider

## Code Review

This Pull Request was thoroughly reviewed and meets all Code Review Standards pertaining to:

- [ ] **Implementation** - satisfied acceptance criteria
- [ ] **Communication** - notified engineers/stakeholders about impactful changes
- [ ] **Error handling** - handled, logged & alerted on errors
- [ ] **Documentation** - wrote readable code & commits, documented unintuitive code
- [ ] **Testing** - unit & E2E test coverage, tested in staging, DB migrations backwards compatible
- [ ] **Security** - sanitized inputs, vetted dependencies, validated/secured API requests, no committed secrets
- [ ] **Performance** - optimized expensive algorithms & database queries

Please select all **applied** standards relevant to this PR.
## Confirmation

- [ ] **Assignee Confirmation** - I (the author) have filled out the template above
- [ ] **Reviewer Confirmation** - I (a/the reviewer) confirm 1) the author has filled out the template and checked the box that says **Assignee Confirmation** 2) I reviewed the code and selected all applicable items in the code review section.

### CODE REVIEW

**Review by DancingAxolotl — APPROVED** [review]
LGTM, added a few minor questions

**Comment by DancingAxolotl** [comment]
@tunguyennnnn Another interesting question, maybe out of scope for this PR. Could we use this approach for workers to have a fallback RPC in case one provider is down?

File: src/services/RpcService.ts [hunk 1]
```diff
@@ -36,14 +63,48 @@ export class RpcService {
...
+    const response = await nodeFetch(rpcUrl, {
+      method: 'post',
+      body: JSON.stringify(body),
+      headers: { 'Content-Type': 'application/json' },
+      timeout: this.requestTimeout,
+    });
+    return response;
+  };
+
+  private sendRpcFallbackRequest = async (
```
[the remark this answers is not in the document]
Comment (tunguyennnnn) [reply src/services/RpcService.ts:75]: yea good call!

**Comment by tunguyennnnn** [comment]
> @tunguyennnnn Another interesting question, maybe out of scope for this PR. Could we use this approach for workers to have a fallback RPC in case one provider is down?

yea definitely.

File: src/services/RpcService.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tunguyennnnn) [reply src/services/RpcService.ts:75]: I'll keep it this way so it's easier to debug and passing logs. I think the module takes the reference of the function as identity of first rpc call. so we have to do something like 
```
getSendRequestFunc (url: rpcUrl)  => async (blockchain: Blockchain, body: RpcPayload) => ...
```

**Pull request merged**